examples/async_load/worker.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+const { promisify } = require('util');
+const sleep = promisify(setTimeout);
+
+module.exports = (async () => {
+  await sleep(500);
+  return () => process._rawDebug('hello from an async loaded CommonJS worker');
```

Comment:
@addaleax ... there's definitely an issue with using `console.log()` and `process.stdout`/`process.stderr` in workers. Specifically, it'll work once or twice in a worker then on subsequent jobs it will just stop. You *should* be able to see that here if you change the `process._rawDebug()` statements to `console.log()` in both workers, then run the example over and over... You'll see something like...

```
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded CommonJS worker
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
hello from an async loaded CommonJS worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded CommonJS worker
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
james@ubuntu:~/nearform/piscina/examples/async_load$ node index
hello from an async loaded ESM worker
hello from an async loaded CommonJS worker
```

I'm only using `process._rawDebug()` here because of the issue.

Reply:
@jasnell I can’t reproduce that (x64 Ubuntu 18.04 with only `process._rawDebug()` → `console.log()`) :/

Reply:
Super odd. I'll come up with a pure node repo and open an issue

Reply:
@jasnell Does setting `useAtomics: false` as a Piscina option make a difference for you?

Reply:
Yeah, setting `useAtomics: false` *appears* to make it work correctly. Also, adding some random delay into each handler also makes it work. Possibly a race condition when multiple workers running at the same time console.log/process.stdout.write at nearly the same time?

Reply:
@jasnell I think it’s the fact that `process.stdout.write()` is an asynchronous operation whose livetime can extend beyond the one of the task itself – the `Atomics.wait()` call in the Worker doesn’t know that and will enter waiting state while more writes might be waiting behind it.

Adjusting the `while (currentTasks === 0) {` condition in `atomicsWaitLoop()` to account for it by adding `&& !process.stdout._writableState.writing && !process.stderr._writableState.writing` might help, but I don’t really want to try that without a reproduction

Reply:
https://github.com/jasnell/piscina/pull/68 should fix this :)